### DISCUSSION TITLE
What's the maximum MTU I can use for Wireguard?

### DISCUSSION THREAD

**Discussion by cornzzy** [erebe/wstunnel#352] [Q&A] [answered]
What's the header size for wss? I don't want all together to be more than 1500.

**Comment by erebe** [comment, marked as answer]
it depends on the data length you forward.
it can vary from 2 to 9 bytes (if you let the default and don't enable masking frame).

If you want to forward udp traffic, as it will be  <1500 bytes, the overhead of websocket is 3 bytes. 

But you also need to account for TLS which is 5bytes. 

So for udp websocket + TLS should be 8 bytes of overhead. 

You can set your MTU to 1450 to be safe.

**Reply by cornzzy** [reply]
Oh I thought readme says 1300. I've tried so many tunnels I've never seen such a small header. This is great. I can use the default MTU of wireguard which is 1420.

**Reply by erebe** [reply]
i mark 1300 just to avoid people complaining. If they see throughput issue setting 1300 should be low enough for them to fix the issue.

Setting it higher require to monitor and understand what is going on and most often it is not the case :/ So 🤷
